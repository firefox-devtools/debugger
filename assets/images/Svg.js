const React = require("react");
import InlineSVG from "svg-inline-react";
const { isDevelopment } = require("devtools-config");

const svg = {
  "angle-brackets": require("./angle-brackets.svg"),
  angular: require("./angular.svg"),
  arrow: require("./arrow.svg"),
  babel: require("./babel.svg"),
  backbone: require("./backbone.svg"),
  blackBox: require("./blackBox.svg"),
  breadcrumb: require("./breadcrumbs-divider.svg"),
  breakpoint: require("./breakpoint.svg"),
  "column-breakpoint": require("./column-breakpoint.svg"),
  "case-match": require("./case-match.svg"),
  choo: require("./choo.svg"),
  close: require("./close.svg"),
  coffeescript: require(`./coffeescript.svg`),
  dojo: require("./dojo.svg"),
  domain: require("./domain.svg"),
  file: require("./file.svg"),
  folder: require("./folder.svg"),
  globe: require("./globe.svg"),
  home: require("./home.svg"),
  javascript: require("./javascript.svg"),
  jquery: require("./jquery.svg"),
  underscore: require("./underscore.svg"),
  lodash: require("./lodash.svg"),
  ember: require("./ember.svg"),
  vuejs: require("./vuejs.svg"),
  "magnifying-glass": require("./magnifying-glass.svg"),
  "arrow-up": require("./arrow-up.svg"),
  "arrow-down": require("./arrow-down.svg"),
  pause: require("./pause.svg"),
  "pause-exceptions": require("./pause-exceptions.svg"),
  plus: require("./plus.svg"),
  preact: require("./preact.svg"),
  aframe: require("./aframe.svg"),
  prettyPrint: require("./prettyPrint.svg"),
  react: require("./react.svg"),
  "regex-match": require("./regex-match.svg"),
  redux: require("./redux.svg"),
  immutable: require("./immutable.svg"),
  resume: require("./resume.svg"),
  settings: require("./settings.svg"),
  stepIn: require("./stepIn.svg"),
  stepOut: require("./stepOut.svg"),
  stepOver: require("./stepOver.svg"),
  subSettings: require("./subSettings.svg"),
  tab: require("./tab.svg"),
  toggleBreakpoints: require("./toggle-breakpoints.svg"),
  togglePanes: require("./toggle-panes.svg"),
  typescript: require("./typescript.svg"),
  "whole-word-match": require("./whole-word-match.svg"),
  worker: require("./worker.svg"),
  "sad-face": require("devtools-mc-assets/assets/devtools/client/themes/images/sad-face.svg"),
  refresh: require("devtools-mc-assets/assets/devtools/client/themes/images/reload.svg"),
  webpack: require("./webpack.svg"),
  node: require("./node.svg"),
  express: require("./express.svg"),
  pug: require("./pug.svg"),
  extjs: require("./sencha-extjs.svg"),
  mobx: require("./mobx.svg"),
  marko: require("./marko.svg"),
  nextjs: require("./nextjs.svg"),
  showSources: require("./showSources.svg"),
  showOutline: require("./showOutline.svg"),
  nuxtjs: require("./nuxtjs.svg"),
  rxjs: require("./rxjs.svg")
};

type SvgType = {
  name: string,
  className?: string,
  onClick?: () => void,
  "aria-label"?: string
};

function Svg({ name, className, onClick, "aria-label": ariaLabel }) {
  if (!svg[name]) {
    const error = `Unknown SVG: ${name}`;
    if (isDevelopment()) {
      throw new Error(error);
    }

    console.warn(error);
    return null;
  }

  className = `${name} ${className || ""}`;
  if (name === "subSettings") {
    className = "";
  }

  const props = {
    className,
    onClick,
    ["aria-label"]: ariaLabel,
    src: svg[name]
  };

  return <InlineSVG {...props} />;
}

Svg.displayName = "Svg";

module.exports = Svg;
