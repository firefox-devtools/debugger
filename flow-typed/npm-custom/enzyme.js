/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

// Using this custom definition of the enzyme module avoids generating some
// flow errors that are unrelated to uses of enzyme itself...
// It would be nice to clean this up.
declare module "enzyme" {
  declare module.exports: any;
}
