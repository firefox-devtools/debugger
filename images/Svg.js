/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

const React = require("react");
import InlineSVG from "svg-inline-react";

const svg = {
  breakpoint: require("./breakpoint.svg"),
  "column-marker": require("./column-marker.svg")
};

type SvgType = {
  name: string,
  className?: string,
  onClick?: () => void,
  "aria-label"?: string
};

function Svg({ name, className, onClick, "aria-label": ariaLabel }: SvgType) {
  if (!svg[name]) {
    const error = `Unknown SVG: ${name}`;
    console.warn(error);
    return null;
  }

  const props = {
    className: `${name} ${className || ""}`,
    onClick,
    "aria-label": ariaLabel,
    src: svg[name]
  };

  return <InlineSVG {...props} />;
}

Svg.displayName = "Svg";

module.exports = Svg;
