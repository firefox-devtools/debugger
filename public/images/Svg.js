const React = require("react");
const InlineSVG = require("svg-inline-react");

const svg = {
  "angle-brackets": require("./angle-brackets.svg"),
  "arrow": require("./arrow.svg"),
  "blackBox": require("./blackBox.svg"),
  "breakpoint": require("./breakpoint.svg"),
  "close": require("./close.svg"),
  "domain": require("./domain.svg"),
  "file": require("./file.svg"),
  "folder": require("./folder.svg"),
  "globe": require("./globe.svg"),
  "magnifying-glass": require("./magnifying-glass.svg"),
  "pause": require("./pause.svg"),
  "pause-exceptions": require("./pause-exceptions.svg"),
  "plus": require("./plus.svg"),
  "prettyPrint": require("./prettyPrint.svg"),
  "resume": require("./resume.svg"),
  "settings": require("./settings.svg"),
  "stepIn": require("./stepIn.svg"),
  "stepOut": require("./stepOut.svg"),
  "stepOver": require("./stepOver.svg"),
  "subSettings": require("./subSettings.svg"),
  "toggleBreakpoints": require("./toggle-breakpoints.svg"),
  "worker": require("./worker.svg"),
  "sad-face": require("./sad-face.svg")
};

module.exports = function(name, props) { // eslint-disable-line
  if (!svg[name]) {
    throw new Error("Unknown SVG: " + name);
  }
  let className = name;
  if (props && props.className) {
    className = `${name} ${props.className}`;
  }
  if (name === "subSettings") {
    className = "";
  }
  props = Object.assign({}, props, { className, src: svg[name] });
  return React.createElement(InlineSVG, props);
};
