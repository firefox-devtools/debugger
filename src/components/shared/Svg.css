/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

.arrow,
.worker,
.refresh,
.shortcut,
.add-button {
  fill: var(--theme-splitter-color);
}

.folder,
.domain,
.source-icon,
.file {
  background-color: var(--theme-comment);
}

.worker,
.file,
.folder,
.source-icon {
  position: relative;
  top: 2px;
}

.domain,
.worker,
.refresh,
.add-button {
  position: relative;
  top: 1px;
}

.worker svg,
.refresh svg,
.shortcut svg,
.add-button svg {
  width: 15px;
}

img.domain,
img.folder,
img.source-icon {
  width: 15px;
  height: 15px;
}

img.result-item-icon {
  height: 18px;
  width: 18px;
}

img.domain {
  mask: url(/images/domain.svg) no-repeat;
}

img.folder {
  mask: url(/images/folder.svg) no-repeat;
}

img.coffeescript {
  mask: url(/images/coffeescript.svg) no-repeat;
}

img.javascript {
  mask: url(/images/javascript.svg) no-repeat;
}

img.tab {
  mask: url(/images/tab.svg) no-repeat;
}

img.react {
  mask: url(/images/react.svg) no-repeat;
}

img.typescript {
  mask: url(/images/typescript.svg) no-repeat;
}

img.file {
  mask: url(/images/file.svg) no-repeat;
  width: 13px;
  height: 13px;
}

img.domain,
img.folder,
img.file,
img.source-icon {
  mask-size: 100%;
  margin-inline-end: 5px;
  display: inline-block;
}

img.result-item-icon {
  mask-size: 100%;
  margin-inline-end: 15px;
  margin-inline-start: 5px;
  display: inline-block;
}

.refresh svg,
.shortcut svg,
.worker svg {
  margin-inline-end: 5px;
}

img.arrow {
  mask: url(/images/arrow.svg);
  margin-inline-end: 5px;
  margin-top: 3px;
  width: 9px;
  height: 9px;
  padding-top: 9px;
  background: var(--disclosure-arrow);
  mask-size: 100%;
  display: inline-block;
  margin-bottom: 1px;
  transform: rotate(-90deg);
  transition: transform 0.18s ease;
}

html[dir="ltr"] img.arrow {
  transform: rotate(-90deg);
}

html[dir="rtl"] img.arrow {
  transform: rotate(90deg);
}

.arrow svg {
  transition: transform 0.125s ease;
  width: 10px;
  margin-inline-end: 5px;
  transform: rotate(-90deg);
}

html[dir="rtl"] .arrow svg,
.arrow svg:dir(rtl),
.arrow svg:-moz-locale-dir(rtl) {
  transform: rotate(90deg);
}

.arrow.expanded.expanded svg {
  transform: rotate(0deg);
}

.arrow.hidden {
  visibility: hidden;
}

.webpack svg {
  width: 15px;
  height: 15px;
  margin-right: 5px;
  vertical-align: sub;
}

.function svg {
  height: 10px;
  width: 15px;
}

.function path {
  fill: var(--theme-body-color);
}

.angular svg,
.source-icon svg {
  width: 15px;
  height: 15px;
  margin-right: 5px;
  vertical-align: sub;
}

.angular,
.webpack {
  opacity: 0.8;
}

.theme-dark .angular,
.theme-dark .webpack {
  opacity: 0.5;
}

.source-icon svg {
  fill: var(--theme-comment);
}
