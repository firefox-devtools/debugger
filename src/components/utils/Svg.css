.worker,
.folder {
    position: relative;
    top: 2px;
}

.arrow,
.folder,
.domain,
.file,
.worker,
.refresh,
.add-button {
    fill: var(--theme-splitter-color);
}

.domain,
.file,
.worker,
.refresh,
.add-button {
    position: relative;
    top: 1px;
}

.domain svg,
.folder svg,
.worker svg,
.refresh svg,
.add-button svg {
    width: 15px;
    fill: currentColor;
}

.file svg {
    width: 13px;
}

html:not([dir="rtl"]) .file svg,
html:not([dir="rtl"]) .domain svg,
html:not([dir="rtl"]) .folder svg,
html:not([dir="rtl"]) .refresh svg,
html:not([dir="rtl"]) .worker svg {
    margin-right: 5px;
}

html[dir="rtl"] .file svg,
html[dir="rtl"] .domain svg,
html[dir="rtl"] .folder svg,
html[dir="rtl"] .worker svg,
html[dir="rtl"] .refresh svg {
    margin-left: 5px;
}
