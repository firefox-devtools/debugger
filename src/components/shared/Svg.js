/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

// @flow
/**
 * This file maps the SVG React Components in the assets/images directory.
 */

import Svg from "../../../assets/images/Svg";

import "./Svg.css";

export default Svg;
