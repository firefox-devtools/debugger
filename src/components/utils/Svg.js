/**
 * This file maps the SVG React Components in the assets/images directory.
 */
const Svg = require("../../../assets/images/Svg");

require("./Svg.css");

module.exports = Svg;
