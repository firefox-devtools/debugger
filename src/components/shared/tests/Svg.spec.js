/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

import { shallow } from "enzyme";
import Svg from "../Svg";

describe("Svg", () => {
  it("renders with props", () => {
    const wrapper = shallow(
      Svg({
        name: "webpack",
        className: "class",
        onClick: () => {},
        "aria-label": "label"
      })
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("render nothing if invalid svg name", () => {
    const svg = Svg({ name: "reallyRandomNameHere" });
    expect(svg).toBe(null);
  });

  it("calls onClick method on click", () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      Svg({
        name: "webpack",
        className: "class",
        onClick,
        "aria-label": "label"
      })
    );
    wrapper.simulate("click");
    expect(onClick).toBeCalledTimes(1);
  });
});
