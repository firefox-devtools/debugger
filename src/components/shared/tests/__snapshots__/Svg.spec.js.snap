// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Svg renders with props 1`] = `
<i
  aria-label="label"
  className="webpack class"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg></svg>",
    }
  }
  onClick={[Function]}
  src={null}
/>
`;
